third_party/nvfuser/csrc/contiguity.h

```diff
@@ -157,7 +157,7 @@ class ContigIDs : public OptInDispatch {
   ContigIDs(
       const std::vector<IterDomain*>& ids,
       const std::vector<IterDomain*>& root_domain,
-      const std::vector<bool>& root_contiguity,
+      const std::vector<c10::optional<bool>>& root_contiguity,
```

Comment:
Why not `std::optional`?

Reply:
I think python frontend is using `c10::optional`, I am just keeping consistent with that. In the future, we can change all them in a batch.

Reply:
That's a good point. We've moved to c++17 already. So it's just legacy.